crates/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{collections::BTreeMap, fmt};
+
+use move_core_types::resolver::{ModuleResolver, ResourceResolver};
+use move_vm_runtime::{move_vm::MoveVM, session::Session};
+use sui_cost_tables::bytecode_tables::GasStatus;
+use sui_types::{
+    base_types::{ObjectID, SuiAddress, TxContext},
+    error::ExecutionError,
+    messages::{Argument, CallArg, ObjectArg},
+    object::Owner,
+    storage::{ChildObjectResolver, ParentSync, Storage},
+};
+
+use crate::adapter::new_session;
+
+use super::types::*;
+
+pub struct ExecutionContext<
+    'v,
+    's,
+    'a,
+    'b,
```

Comment:
Oh my...

Reply:
Should `'a` and `'b` be `'c` and `'g` respectively?

Reply:
I could take the names the session uses if that makes more sense? Which is `'l` and `'r` lol